site/en/docs/extensions/mv3/tut_analytics/index.md

```diff
@@ -1,166 +1,261 @@
 
-```html
-<button id='button1'>Button 1</button>
-<button id='button2'>Button 2</button>
-<button id='button3'>Button 3</button>
+```js
+const GA_ENDPOINT = "https://www.google-analytics.com/mp/collect";
+const MEASUREMENT_ID = `G-...`;
+const API_SECRET = `...`;
+const DEFAULT_ENGAGEMENT_TIME_IN_MSEC = 100;
+
+fetch(
+`${GA_ENDPOINT}?measurement_id=${MEASUREMENT_ID}&api_secret=${API_SECRET}`,
+  {
+    method: "POST",
+    body: JSON.stringify({
+      client_id: getOrCreateClientId(),
+      events: [
+        {
+          name: "button_clicked",
+          params: {
+            session_id: await this.getOrCreateSessionId(),
+            engagement_time_msec" DEFAULT_ENGAGEMENT_TIME_IN_MSEC,
+            id: "my-button",
+          },
+        },
+      ],
+    }),
+  }
+);
 ```
 
-Write a function that sends click events to Google Analytics:
+The event will be displayed as follows in the Google Analytics real-time report.
+
+{% Img src="image/6hHqS5auVgWhN0cQNQztaJx5w4M2/AQqSJb9VqalWfrQ6xSXM.png", alt="Realtime events in Google Analytics. ", width="396", height="414" %}
+
+
+## Tracking page views in tabs, popup and sidepanel  {: #toc-tracking-pageviews }
+The Google Analytics Measurement Protocol supports a special `page_view` event for tracking page views. Use this to track users visiting your popup pages, tab pages or sidepanel. The `page_view` event also requires the `page_title` and `page_location` parameters. The following example fires a page view event at the document [`load`](https://developer.mozilla.org/docs/Web/API/Window/load_event) event for an extension popup.:
+
+{% Label %}popup.js:{% endLabel %}
 
 ```js
-function trackButton(e) {
-  _gaq.push(['_trackEvent', e.target.id, 'clicked']);
-};
+window.addEventListener("load", () => {
+  ${GA_ENDPOINT}?measurement_id=${MEASUREMENT_ID}&api_secret=${API_SECRET}`,
+  {
+    method: "POST",
+    body: JSON.stringify({
+      client_id: getOrCreateClientId(),
+      events: [
+        {
+          name: "page_view",
+          params: {
+            session_id: await this.getOrCreateSessionId(),
+            engagement_time_msec" DEFAULT_ENGAGEMENT_TIME_IN_MSEC,
+            page_title: document.title, 
+            page_location: document.location.href
+          },
+        },
+      ],
+    }),
+  }
+});
 ```
 
-And use it as an event handler for each button's click:
+The `popup.js` script needs to be imported in your popup’s html file and should run before any other script is executed:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Analytics Demo Popup</title>
+    <script src="./popup.js" type="module"></script>
+  </head>
+  <body>
+    <h1>Analytics Demo</h1>
+  </body>
+</html>
+```
+
+The popup view will be displayed like any other page view in the Google Analytics real-time report:
```

Comment:
```suggestion
The popup view will be displayed like any other page view in the Google Analytics Realtime report:
```
Making this suggestions based on https://support.google.com/analytics/answer/9271392?hl=en